artemis-commons/src/main/java/org/apache/activemq/artemis/utils/critical/CriticalMeasure.java

```diff
@@ -17,28 +17,39 @@
 
 package org.apache.activemq.artemis.utils.critical;
 
-import org.jboss.logging.Logger;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
 
-public class CriticalMeasure {
+final class CriticalMeasure {
```

Comment:
why no longer public?

also on final, i assume this isn't for performance as already raised before making a class final does not improve perf.

Reply:
Exactly , It is mostly a design decision :)

Reply:
I'd rather this was left as was then.

Reply:
Why?

Reply:
Why change it? Else we end up just flip flopping this based on people's own design pref. It's out of scope of the change eg it's not needed

Reply:
It is a good point for sure, but I agree only on the second half: is out of scope for the change so I can push a different PR just with it (and for the atomic changes too I can say).
For the first half I do not agree: is not a personal taste, but the explicit declaration of the original intent. That class wasn't designed to be extended or used in other points in the code.

Reply:
The intention of the CriticalMeasure is to be eventually exposed to people using embedded systems... or to eventually to become like a profiler where we could expose through management. So, keep it public.